test/unit/RelayServer.test.ts

```diff
@@ -0,0 +1,232 @@
+    let mockDependencies: ServerDependencies;
+
+    beforeEach(() => {
+        fakeManagerKeyManager = createStubInstance(KeyManager, {
+            getAddress: stub(),
+        });
+        fakeManagerKeyManager.getAddress.returns('fake_address');
+
+        fakeWorkersKeyManager = createStubInstance(KeyManager, {
+            getAddress: stub(),
+        });
+        fakeWorkersKeyManager.getAddress.returns('fake_address');
+
+        fakeContractInteractor = createStubInstance(ContractInteractor);
+
+        fakeTxStoreManager = createStubInstance(TxStoreManager);
+
+        mockDependencies = {
+            managerKeyManager: fakeManagerKeyManager,
+            workersKeyManager: fakeWorkersKeyManager,
+            contractInteractor: fakeContractInteractor,
+            txStoreManager: fakeTxStoreManager,
+        };
+    });
+
+    describe('constructor', () => {
+        it('should call getAddress of key managers', () => {
+            const server = new RelayServer({
+                logLevel: 1
+            }, mockDependencies);
+
+            expect(server, 'Is instance of RelayServer').to.be.instanceOf(
+                RelayServer
+            );
+
+            expect(fakeManagerKeyManager.getAddress, 'Calls manager getAddress').to
+                .have.been.called;
+            expect(fakeWorkersKeyManager.getAddress, 'Calls workers getAddress').to
+                .have.been.called;
+        });
+    });
+
+    describe('getMaxPossibleGas', async () => {
+        const fakeRelayTransactionRequest: RelayTransactionRequest = {
+            relayRequest: {
+                relayData: {
+                    gasPrice: '0',
+                    callForwarder: 'fake_address',
+                } as RelayData,
+                request: {
+                    to: 'fake_address',
+                    data: 'fake_data',
+                    gas: '1',
+                } as ForwardRequest,
+            },
+            metadata: stubInterface<RelayMetadata>(),
+        };
+
+        const examplePrice = toBN(10).pow(toBN(14));
+
+        const setMaxGas = (price: number) => fakeContractInteractor.estimateRelayTransactionMaxPossibleGasWithTransactionRequest.returns(
+            Promise.resolve(price)
+        );
+
+        it('should return gas price as is if sponsoredTxFee = 0', async () => {
+            const expectedGasPrice = examplePrice;
+            setMaxGas(expectedGasPrice.toNumber());
+
+            const server = new RelayServer({}, mockDependencies);
+            const actualGasPrice = await server.getMaxPossibleGas(
+                fakeRelayTransactionRequest,
+                false
+            );
+
+            expect(actualGasPrice.eq(expectedGasPrice)).to.be.true;
+        });
+
+        afterEach(() => {
+            Sinon.restore();
+        });
+
+        it('should throw `User agreed to spend lower than what the transaction may require.` if tokenAmount < gas + gas * fee', async () => {
```

Comment:
If you decide to use the string constants, we could replace it here

Reply:
done